libpod/container_internal_linux.go

```diff
@@ -2445,6 +2445,12 @@ func (c *Container) getOCICgroupPath() (string, error) {
 		// slice:prefix:name
 		systemdCgroups := fmt.Sprintf("%s:libpod:%s", path.Base(c.config.CgroupParent), c.ID())
 		logrus.Debugf("Setting CGroups for container %s to %s", c.ID(), systemdCgroups)
+		// If user is rootless and no valid systemd user session is present then break early.
+		if rootless.IsRootless() {
+			if !utils.CheckifSystemdSessionisValid(rootless.GetRootlessUID()) {
+				return "", errors.Wrapf(define.ErrInternal, "invalid systemd user session")
```

Comment:
Can we have a warning instead of hard failure and let runtime handle the failures ?